### PR TITLE
terraform 0.7.6

### PR DESCRIPTION
**Pull request by cblecker**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by zmwangx** [comment]
Build failed on Yosemite. https://bot.brew.sh/job/Homebrew%20Core/9399/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_terraform/

**Comment by cblecker** [comment]
@zmwangx --
I bumped the dependency versions, as well as added the parallelization and timeout options per the Makefile (ref: https://github.com/hashicorp/terraform/blob/v0.7.6/Makefile#L43)

Build now passes on all platforms. 😃

**Comment by MikeMcQuaid** [comment]
> I bumped the dependency versions, as well as added the parallelization and timeout options to the test per the Makefile (ref: https://github.com/hashicorp/terraform/blob/v0.7.6/Makefile#L43)

@cblecker Out of interest: could we just use the `Makefile`?

**Comment by cblecker** [comment]
@MikeMcQuaid From what I can tell, yes, it might be possible to change the Formula around to use the Makefile instead of pulling out commands into the Formula itself. Which is more preferred from the Homebrew standpoint? Is this worth me digging into?

**Comment by MikeMcQuaid** [comment]
Yeh, using `make` would be preferred for us, thanks 👍

**Comment by cblecker** [comment]
@MikeMcQuaid I've changed the Formula to now use the Makefile. Builds pass on all platforms.

**Comment by cblecker** [comment]
An added benefit I've noticed is that by using the release build procedure in the Makefile, the binary is about 50M smaller (123M -> 76M). This was definitely the way to go 👍

File: Formula/terraform.rb [hunk 1]
```diff
@@ -52,21 +54,27 @@ def install
...
+
     cd terrapath do
       # v0.6.12 - source contains tests which fail if these environment variables are set locally.
       ENV.delete "AWS_ACCESS_KEY"
       ENV.delete "AWS_SECRET_KEY"
 
-      terraform_files = `go list ./...`.lines.map { |f| f.strip unless f.include? "/vendor/" }.compact
-      system "go", "test", *terraform_files
+      # Runs format check and test suite via makefile
+      ENV.deparallelize { system "make", "test", "vet" }
```
Comment (MikeMcQuaid) [inline Formula/terraform.rb]: Could you report an issue upstream about this needing to be run with `-j1`?

File: Formula/terraform.rb [hunk 2]
```diff
@@ -52,21 +54,27 @@ def install
...
-        "-arch", arch,
-        "-os", "darwin",
-        "-output", "bin/terraform-{{.Dir}}", *terraform_files
-      bin.install "bin/terraform-terraform" => "terraform"
+      ENV["XC_OS"] = "darwin"
+      ENV["XC_ARCH"] = arch
+      system "make", "bin"
+
+      # Install release binary
+      bin.install "pkg/darwin_#{arch}/terraform"
```
Comment (MikeMcQuaid) [inline Formula/terraform.rb]: Would be great to request/add a `make install` upstream.

Comment (cblecker) [inline Formula/terraform.rb]: The build places the file (in addition to the above place) in the `bin/` relative directory, as well as the `$GOPATH/bin/` directory. So while they're in the right places in the tmp files, homebrew doesn't pick up on them to actually move them over to the Cellar without the install command.

Any leads as to why?

File: Formula/terraform.rb (hunk 1 above)
Comment (cblecker) [inline Formula/terraform.rb]: I've spent some time looking at this today, and I'm still not sure what exactly is breaking. I've tried multiple times on my local system with the `-j4` MAKEFILES env variable, similar to the Jenkins environment, and it works every time. So I must be missing something about the way @BrewTestBot is building this, as it's hitting the same race condition every single time that my local machine isn't when attempting to build the bottle.

I don't mind reporting this issue, but I don't feel I have enough diagnostic details to.

Comment (MikeMcQuaid) [inline Formula/terraform.rb]: Doing `-j20` or `-j` may help reveal this problem more easily; generally any time a `Makefile` has a parellelisation issue like this it's considered a bug. That said, I appreciate you've spent a bunch of time here so I'll pull this as-is but a report upstream would still be great (and I'm happy to help, there).

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**